pkg/operator/controller/nodeobservabilityrun/mco.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeobservabilityruncontroller
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/openshift/node-observability-operator/api/v1alpha1"
+)
+
+const (
+	MCOKind       = "NodeObservabilityMachineConfig"
+	MCOApiVersion = "nodeobservability.olm.openshift.io/v1alpha1"
+	MCOName       = "nodeobservabilitymachineconfig-run"
+)
+
+// ensureMCO ensures that the node observability machineconfig is created
+// Returns a Boolean value indicating whether it exists, a pointer to the
+// daemonset and an error when relevant
+func (r *NodeObservabilityRunReconciler) ensureMCO(ctx context.Context) (bool, *v1alpha1.NodeObservabilityMachineConfig, error) {
+	desired := r.desiredMCO()
+	nameSpace := types.NamespacedName{Namespace: "", Name: desired.Name}
+	exist, current, err := r.currentMCO(ctx, nameSpace)
+	if err != nil {
+		return false, nil, fmt.Errorf("failed to get NodeObservabilityMachineConfig: %v", err)
+	}
+	if !exist {
+		if err := r.createMCO(ctx, desired); err != nil {
+			return false, nil, err
+		}
+		return r.currentMCO(ctx, nameSpace)
+	}
```

Comment:
Would there be a need to check if someone forcefully edited the MCO, and update it to the content expected by the nodeobservabilityrun controller?

Reply:
I just followed the normal pattern I used for all other objects. 
What were you thinking of here ?

Reply:
Discussed with @sherine-k - for tech preview we won't worry about this, but address this or future releases